docs/hugo/content/contributing/add-a-new-code-generated-resource/write-a-test.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Write a test for the resource
+linktitle: Write a test
+weight: 40
+---
+
+Every new resource should have a handwritten test as there is always the possibility of a resource behaving in a subtly different way than expected. This also applies to _new versions_ of existing resources.
+
+All these tests live in the [`v2/internal/controllers`](https://github.com/Azure/azure-service-operator/tree/main/v2/internal/controllers). There's a readme in that folder that describes the test structure and naming conventions.
+
+For each test there's a recording in the `recordings` folder that records the HTTP interactions between the test and Azure allowing for rapid replay for future runs.
```

Comment:
Maybe briefly describe the vcr/recording stuff (and/or link to another doc about it) as you mention "recording" here but no prior context?